### PR TITLE
add the extension ListTuplePuns (GHC proposal 475)

### PR DESCRIPTION
**Pull request by tek**
Upcoming extension, see https://gitlab.haskell.org/ghc/ghc/-/merge_requests/8820

### CODE REVIEW

**Comment by ulysses4ever** [comment]
Thanks for the contribution!

CI fails in the usual place: you have to update that hash whenever add an extension.

**Comment by Mikolaj** [comment]
This time it failed to a usual GHA outage, so I've restarted failed jobs.

**Comment by Mikolaj** [comment]
@tek: is this ready for merge? If so, could you rebase, remove the `Draft` mark and set the review_needed label? (Please let me know if any of these doesn't work.)

**Comment by tek** [comment]
the GHC MR is still not approved, though Vlad is pushing for it, so might be a few days.
I only rebased for now.

**Comment by tek** [comment]
ok I guess this doesn't have to wait for the GHC MR to be merged!

**Comment by tek** [comment]
@Mikolaj I have no label permissions, though

**Comment by Mikolaj** [comment]
How about now?

**Comment by tek** [comment]
thanks!

**Review by geekosaur — APPROVED** [review]

**Comment by tek** [comment]
hey @Mikolaj, would you be so kind and merge this now?

File: changelog.d/pr-8854
Comment (ulysses4ever) [inline changelog.d/pr-8854:4]: Not sure it's a significant change for Cabal: we simply follow GHC here (it may be deemed significant on GHC side).

File: changelog.d/pr-8854
Comment (ulysses4ever) [inline changelog.d/pr-8854:8]: Consider using Markdown markup. E.g. `ListTuplePuns`.

**Review by ulysses4ever — APPROVED** [review]

**Comment by ulysses4ever** [comment]
Feel free to proceed with the labels, as described here: https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#github-pull-request-conventions

**Comment by Mikolaj** [comment]
@tek: apparently, a conflict with the master branch popped up at the last moment. Once you resolve it, merging should resume (probably immediately, but at worst in 2 days).

**Pull request merged**
